osu.Game/Screens/Edit/Compose/Components/ComposeBlueprintContainer.cs

```diff
@@ -70,6 +72,50 @@ protected override void LoadComplete()
             }
         }
 
+        protected override bool OnKeyDown(KeyDownEvent e)
+        {
+            if (e.ControlPressed)
+            {
+                switch (e.Key)
+                {
+                    case Key.Left:
+                        moveSelection(new Vector2(-1, 0));
+                        return true;
+
+                    case Key.Right:
+                        moveSelection(new Vector2(1, 0));
+                        return true;
+
+                    case Key.Up:
+                        moveSelection(new Vector2(0, -1));
+                        return true;
+
+                    case Key.Down:
+                        moveSelection(new Vector2(0, 1));
+                        return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Move the current selection spatially by the specified delta, in gamefield coordinates (ie. the same coordinates as the blueprints).
+        /// </summary>
+        /// <param name="delta"></param>
+        private void moveSelection(Vector2 delta)
+        {
+            var firstBlueprint = SelectionHandler.SelectedBlueprints.FirstOrDefault();
+
+            if (firstBlueprint == null)
+                return;
+
+            // convert to game space coordinates
+            delta = firstBlueprint.ToScreenSpace(delta) - firstBlueprint.ToScreenSpace(Vector2.Zero);
+
+            SelectionHandler.HandleMovement(new MoveSelectionEvent(firstBlueprint, firstBlueprint.ScreenSpaceSelectionPoint + delta));
```

Comment:
Using `HandleMovement()` confuses me a bit here. The previous usages of that method at `BlueprintContainer` were shared by the compose and timeline blueprint containers. The compose container does positional snapping, and the timeline container does time snapping. The xmldoc of `HandleMovement()` also mentions time snapping, although there's no snapping ever happening here, I don't think?

In this context the method is being used for its "implementation-specific effects" of positional movement, I believe. It might just be that it's the xmldoc that's confusing and that snapping mention needs to be trimmed, but as is I'm a bit lost as to how an implementation of `HandleMovement()` is supposed to look like. It also seems like time and position concerns are intermingled too much, although that's the pre-existing state before this PR, I suppose.

Reply:
You raise a good point. I may just be able to apply the changes directly to the `HitObject`s and also bypass the coordinate space conversion as well.

Reply:
I've updated the implementation to avoid using `HandleMovement` (see https://github.com/peppy/osu/tree/add-editor-nudgin-support-v2), but the one issue that I found is it no longer bounds it to the playfield area (that logic is currently in `OsuSelectionHandler.moveSelection`).

I do actually thing the existing method may be for the best...

The `StartTime` mention in the xmldoc is referring to scrolling playfields, which are automatically handling drag movement to change hitobject time, rather than position. I'm not sure how to make this read better but maybe it should be changed to avoid confusion.

Reply:
Okay, yeah, with the mention that the start time changing applies to scrolling rulesets this makes more sense.

The coordinate space precision does suck quite a bit, though, since I'm seeing floating point accumulation errors with the current setup... For instance, something like

![2021-03-26-214251_983x285_scrot](https://user-images.githubusercontent.com/20418176/112690587-52bae100-8e7c-11eb-8f65-eb2a5ebde5ea.png)

turns to

![2021-03-26-214319_960x263_scrot](https://user-images.githubusercontent.com/20418176/112690599-58b0c200-8e7c-11eb-889e-196a2f693492.png)

after a few nudges. Note the fractional part drifting off ever so slightly (happens because of accumulating FP/representation error - the difference between them is not a perfect integer as perhaps expected). But I'm not super sure this can be fully fixed even if the space transformation is removed, because any series of repeated non-power-of-2 subtractions can do the same thing, as long as the backing type is a `float`...

Reply:
Hmm, I think it should be fine to round to `int`. The whole idea of this nudge feature is to give mappers the int layout precision they apparently need, after all.

Reply:
Well, I guess that's not actually too simple to do with the current coordinate space change logic. Definitely not idea, but can't think of a better way for now (without some restructuring of how positional changes are done in general).

Reply:
Probably fine to get it in as-is and fix in the future? I don't think sub-pixel errors are going to matter much to most people for now.

Reply:
Yeah, I'd say it's one case where it'd fine until someone notices it.